servicetalk-grpc-api/src/main/java/io/servicetalk/grpc/api/GrpcMessageEncoding.java

```diff
@@ -1,37 +0,0 @@
-/*
- * Copyright © 2019 Apple Inc. and the ServiceTalk project authors
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *   http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package io.servicetalk.grpc.api;
-
-/**
- * API for <a href="https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#message-encoding"> message
- * coding schemes</a>.
- */
-public interface GrpcMessageEncoding {
```

Comment:
I am on the fence about removing this. I guess we could keep this around and use this interface for gRPC (delegating to the HTTP codec), allowing for logic separation if we need it in the future. @idelpivnitskiy thoughts?

Reply:
Can you please clarify why you expect the need for gRPC-specific type?

Reply:
I can't think of something, that's why I removed it. 
I am just considering whether keeping it allows for some flexibility on doing changes to one without affecting both.

Reply:
After thinking about it more, I'm agreed, it may be safer to have protocol-specific types that delegate the actual compression/decompression work to protocol-agnostic types. 
We may have `servicetalk-content-encoding-api` module that will have `ContentCodec` class and identity, gzip, deflate implementations. Later we can create `servicetalk-content-encoding-netty` for netty-based implementations.

Then we may have `HttpContentEncoding` for `http-api` and `GrpcMessageEncoding` for `grpc` modules. Each will delegate to the necessary API of `ContentCodec` + have an additional method to set up required header value:

For `HttpContentEncoding`:
```java
default HttpContentEncoding populateHeaders(final HttpHeader headers) {
    headers.add(CONTENT_ENCODING, name());
    headers.add(VARY, CONTENT_ENCODING);
}
```

For `GrpcMessageEncoding` (should it extend `HttpContentEncoding`?):
```java
@Override
default GrpcMessageEncoding populateHeaders(final HttpHeader headers) {
    headers.set(GRPC_MESSAGE_ENCODING_KEY, name());
}
```
WDYT?